src/components/NewCommentForm.tsx

```diff
@@ -1,103 +1,222 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import classNames from 'classnames';
+import { Comment } from '../types/Comment';
+import { addComment } from '../api/api';
+
+type Props = {
+  postId: number;
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [noName, setNoName] = useState(false);
```

Comment:
mb would be better like
```suggestion
  const [hasName, setHasName] = useState(false);
```